pkg/requestgen/generator.go

```diff
@@ -56,32 +58,58 @@ func RequestForMetadata(meta metadata.Metadata) (*manager.CertificateRequestBund
 		}
 	}
 
-	commonName, err := expand(meta, attrs[csiapi.CommonNameKey])
-	if err != nil {
-		return nil, fmt.Errorf("%q: %w", csiapi.CommonNameKey, err)
+	var request = &x509.CertificateRequest{}
+	if lSubjStr, ok := attrs[csiapi.LiteralSubjectKey]; ok && len(lSubjStr) > 0 {
+		lSubjStr, err = expand(meta, lSubjStr)
+		if err != nil {
+			return nil, fmt.Errorf("%q: %w", csiapi.LiteralSubjectKey, err)
+		}
+		request.RawSubject, err = cmpki.ParseSubjectStringToRawDerBytes(lSubjStr)
+		if err != nil {
+			return nil, fmt.Errorf("%q: %w", csiapi.LiteralSubjectKey, err)
+		}
+	} else {
+		request.Subject = pkix.Name{}
+		request.Subject.CommonName, err = expand(meta, attrs[csiapi.CommonNameKey])
+		if err != nil {
+			return nil, fmt.Errorf("%q: %w", csiapi.CommonNameKey, err)
+		}
+		if len(attrs[csiapi.SerialNumberKey]) > 0 {
+			request.Subject.SerialNumber = attrs[csiapi.SerialNumberKey]
+		}
+		for k, v := range map[*[]string]string{
+			&request.Subject.Organization:       csiapi.OrganizationsKey,
+			&request.Subject.OrganizationalUnit: csiapi.OrganizationalUnitsKey,
+			&request.Subject.Country:            csiapi.CountriesKey,
+			&request.Subject.Province:           csiapi.ProvincesKey,
+			&request.Subject.Locality:           csiapi.LocalitiesKey,
+			&request.Subject.StreetAddress:      csiapi.StreetAddressesKey,
+			&request.Subject.PostalCode:         csiapi.PostalCodesKey,
+		} {
+			if len(attrs[v]) > 0 {
+				var e, err = expand(meta, attrs[v])
+				if err != nil {
+					return nil, fmt.Errorf("%q: %w", v, err)
+				}
+				*k = strings.Split(e, ",")
```

Comment:
suggestion: `strings.Split` might not be suitable here, since several of these fields can reasonably contain commas.

I remember that you commented on a PR implementing a similar feature in cert-manager: https://github.com/cert-manager/cert-manager/pull/4502#issuecomment-1371155923

I think we're likely to need a similar approach here as I suggested in there. CSV parsing seems like a good solution.

Example test string:

```text
`"1725 Slough Avenue, Suite 200, Scranton Business Park","10 Downing Street, Westminster",Something Else`
```

Should produce exactly 3 street address entries

Reply:
@cornfeedhobo

Reply:
Thank you for letting us know!